azure-quantum/azure/quantum/optimization/problem.py

```diff
@@ -124,12 +127,46 @@ def add_term(self, c: Union[int, float], indices: List[int]):
         self.terms.append(Term(indices=indices, c=c))
         self.uploaded_blob_uri = None
 
-    def add_terms(self, terms: List[Term]):
-        """Adds a list of terms to the `Problem` representation
+    def add_terms(self, terms: List[Term],
+                  type: str = None, c: Union[int, float] = 1):
+        """Adds an optionally grouped list of monomial terms 
+        to the `Problem` representation
 
         :param terms: The list of terms to add to the problem
+        :param type: Type of grouped term being added, if applicable
+        :param c: Weight of grouped term, if applicable
+        """
+        if type:
+            # Grouped term
+            if type == 'na':
+                gtype = GroupType.combination
+            elif type == 'slc':
+                gtype = GroupType.squared_linear_combination
+            else:
+                raise Exception("Unsupported group type {}.".format(type))
+            self.terms.append(GroupedTerm(gtype, terms, c=c))
+        else:
+            # List of ungrouped monomial terms
+            self.terms += terms
+        self.uploaded_blob_uri = None
+    
+    def add_slc_term(self,
+                     terms: List[Tuple[Union[int, float], Union[int, Type[None]]]],
```

Comment:
I remember we discussed to allow both a list of terms as well as the list of tuples. 
Also i think  Union[int, Type[None]] can be replaced with Optional[int]

Reply:
The add_slc_term accepts either a list of tuples or a list of Term objects (new); the add_terms function accepts a list of Term objects and optionally groups them according to a type specification (including SLC).